[package] xmlsec/1.3.6: MinGW on Linux: header file conflicts between OpenSSL and Windows

### Description

When trying to compile the recipe `xmlsec/1.3.6` with the `openssl/1.1.1w` dependency, there is a header conflict between OpenSSL with Windows headers included by xmlsec when trying to cross-compile with MinGW on Linux the library for Windows x86 (32-bit) target:

```
In file included from /usr/share/mingw-w64/include/windows.h:95,
                 from /home/amachado/.conan2/p/b/xmlsec164b7f37bd9d/b/src/include/xmlsec/private.h:36,
                 from /home/amachado/.conan2/p/b/xmlsec164b7f37bd9d/b/src/src/openssl/x509.c:39:
/home/amachado/.conan2/p/b/opens49f1201400053/p/include/openssl/x509v3.h:140:9: error: expected specifier-qualifier-list before '(' token
  140 |         X509_NAME *directoryName;
      |         ^~~~~~~~~
/home/amachado/.conan2/p/b/opens49f1201400053/p/include/openssl/x509v3.h:179:5: error: expected specifier-qualifier-list before '(' token
  179 |     X509_NAME *dpname;
      |     ^~~~~~~~~
```

As workaround  and based the original [msvc makefile](https://github.com/lsh123/xmlsec/blob/f48168188a9d0ecda4a19b3f5586860b00d81b69/win32/Makefile.msvc#L365-L375) I changed the `xmlsec` recipe to include the define `WIN32_LEAN_AND_MEAN` when the compiler is mingw that I will suggest in a PR:

```diff
diff --git a/recipes/xmlsec/all/conanfile.py b/recipes/xmlsec/all/conanfile.py
index f6e3b2305f..feb4125a5a 100644
--- a/recipes/xmlsec/all/conanfile.py
+++ b/recipes/xmlsec/all/conanfile.py
@@ -46,6 +46,10 @@ class XmlSecConan(ConanFile):
     def _settings_build(self):
         return getattr(self, "settings_build", self.settings)
 
+    @property
+    def _is_mingw(self):
+        return self.settings.compiler == "gcc" and self.settings.os == "Windows"
+
     def config_options(self):
         if self.settings.os == "Windows":
             del self.options.fPIC
@@ -105,6 +109,9 @@ class XmlSecConan(ConanFile):
             tc = AutotoolsToolchain(self)
             if not self.options.shared:
                 tc.extra_defines.append("XMLSEC_STATIC")
+            if self._is_mingw:
+                tc.extra_defines.append("WIN32_LEAN_AND_MEAN")
+
             yes_no = lambda v: "yes" if v else "no"
             tc.configure_args.extend([
                 "--enable-crypto-dl=no",
```

I don't know if there is an alternative suggestion to fix this issue, but this approach resolved this conflict when compiling.

### Package and Environment Details

* Package Name/Version: **xmlsec/1.3.6**
* Operating System+version: **Linux Ubuntu 24.04.1 LTS**
* Compiler+version: **i686-w64-mingw32-gcc (GCC) 13-win32**
* Conan version: **conan 2.12.1**
* Python version: **Python 3.12.3**


### Conan profile

Profile Host: **windows-x86**:
```
[settings]
os=Windows
arch=x86
build_type=Release
compiler=gcc
compiler.cppstd=11
compiler.libcxx=libstdc++11
compiler.version=6.3

[conf]
tools.cmake.cmaketoolchain:user_toolchain=["{{profile_dir}}/windows-x86.toolchain.cmake"]
tools.build:compiler_executables={"c":"i686-w64-mingw32-gcc","cpp":"i686-w64-mingw32-g++","rc":"i686-w64-mingw32-windres"}

[buildenv]
CFLAGS=-m32 -march=pentium-mmx
CXXFLAGS=-m32 -march=pentium-mmx
LDFLAGS=-m32

AS=i686-w64-mingw32-as
LD=i686-w64-mingw32-ld
CC=i686-w64-mingw32-gcc
CXX=i686-w64-mingw32-g++
AR=i686-w64-mingw32-ar
NM=i686-w64-mingw32-nm
STRIP=i686-w64-mingw32-strip
OBJCOPY=i686-w64-mingw32-objcopy
OBJDUMP=i686-w64-mingw32-objdump
RANLIB=i686-w64-mingw32-ranlib
RC=i686-w64-mingw32-windres
WINDRES=i686-w64-mingw32-windres
UPX_PROGRAM=upx
```

**windows-x86.toolchain.cmake**:
```
set(CMAKE_SYSTEM_NAME Windows)
set(CMAKE_SYSTEM_PROCESSOR x86)

set(CMAKE_C_COMPILER i686-w64-mingw32-gcc)
set(CMAKE_CXX_COMPILER i686-w64-mingw32-g++)
set(CMAKE_ASM_COMPILER i686-w64-mingw32-as)
set(CMAKE_AR i686-w64-mingw32-ar)
set(CMAKE_NM i686-w64-mingw32-nm)
set(CMAKE_RANLIB i686-w64-mingw32-ranlib)
set(CMAKE_STRIP i686-w64-mingw32-strip)
set(CMAKE_OBJCOPY i686-w64-mingw32-objcopy)
set(CMAKE_OBJDUMP i686-w64-mingw32-objdump)
set(CMAKE_RC_COMPILER i686-w64-mingw32-windres)
```

Profile Build: **default**:
```
[settings]
arch=x86_64
build_type=Release
compiler=gcc
compiler.cppstd=gnu17
compiler.libcxx=libstdc++11
compiler.version=13
os=Linux
[conf]
tools.system.package_manager:mode=install
tools.system.package_manager:sudo=True
tools.system.package_manager:tool=apt-get
```

### Steps to reproduce

1. Install MinGW i686 compilers: `sudo apt install gcc-mingw-w64-i686 g++-mingw-w64-i686`

2. In a folder like `/tmp/xmlsec_test/` put the profile and toolchain files `windows-x86`  `windows-x86.toolchain.cmake` as described  in the **Conan profile** section, and a `conanfile.txt` with this content:

```
[requires]
xmlsec/1.3.6
openssl/1.1.1w
[generators]
CMakeDeps
CMakeToolchain
[layout]
cmake_layout
```

3. Run the following command:
`conan install . -r conancenter -pr:h=./windows-x86 -pr:b=default  -o "libxml2/*:include_utils=False" --build=missing`

Notice: we had to set `libxml2/*:include_utils=False` because that dependency recipe has also an issue with MinGW with this default enabled option, raising a build error of missing `config.h`


### Logs

<details><summary>Click to expand build log</summary>

```
======== Input profiles ========
Profile host:
[settings]
arch=x86
build_type=Release
compiler=gcc
compiler.cppstd=11
compiler.libcxx=libstdc++11
compiler.version=6.3
os=Windows
[options]
libxml2/*:include_utils=False
[conf]
tools.build:compiler_executables={'c': 'i686-w64-mingw32-gcc', 'cpp': 'i686-w64-mingw32-g++', 'rc': 'i686-w64-mingw32-windres'}
tools.cmake.cmaketoolchain:user_toolchain=['/tmp/xmlsec_test/windows-x86.toolchain.cmake']
tools.system.package_manager:mode=install
tools.system.package_manager:sudo=True
tools.system.package_manager:tool=apt-get
[buildenv]
CFLAGS=-m32 -march=pentium-mmx
CXXFLAGS=-m32 -march=pentium-mmx
LDFLAGS=-m32
AS=i686-w64-mingw32-as
LD=i686-w64-mingw32-ld
CC=i686-w64-mingw32-gcc
CXX=i686-w64-mingw32-g++
AR=i686-w64-mingw32-ar
NM=i686-w64-mingw32-nm
STRIP=i686-w64-mingw32-strip
OBJCOPY=i686-w64-mingw32-objcopy
OBJDUMP=i686-w64-mingw32-objdump
RANLIB=i686-w64-mingw32-ranlib
RC=i686-w64-mingw32-windres
WINDRES=i686-w64-mingw32-windres
UPX_PROGRAM=upx

Profile build:
[settings]
arch=x86_64
build_type=Release
compiler=gcc
compiler.cppstd=gnu17
compiler.libcxx=libstdc++11
compiler.version=13
os=Linux
[conf]
tools.system.package_manager:mode=install
tools.system.package_manager:sudo=True
tools.system.package_manager:tool=apt-get


======== Computing dependency graph ========
xmlsec/1.3.6: Not found in local cache, looking in remotes...
xmlsec/1.3.6: Checking remote: conancenter
Connecting to remote 'conancenter' anonymously
xmlsec/1.3.6: Downloaded recipe revision 57dd2c8589a0137ff8b9fd65fcb76348
Graph root
    conanfile.txt: /tmp/xmlsec_test/conanfile.txt
Requirements
    libiconv/1.17#1ae2f60ab5d08de1643a22a81b360c59 - Cache
    libxml2/2.13.4#72b3bfba0c37dfe22f20c587cc468f4c - Cache
    openssl/1.1.1w#1457e590534e0b6963d32e0d5ab44eb5 - Cache
    xmlsec/1.3.6#57dd2c8589a0137ff8b9fd65fcb76348 - Downloaded (conancenter)
    zlib/1.3.1#b8bc2603263cf7eccbd6e17e66b0ed76 - Cache
Build requirements
    autoconf/2.71#f9307992909d7fb3df459340f1932809 - Cache
    automake/1.16.5#058bda3e21c36c9aa8425daf3c1faf50 - Cache
    gnu-config/cci.20210814#dc430d754f465e8c74463019672fb97b - Cache
    libtool/2.4.7#08316dad5c72c541ed21e039e4cf217b - Cache
    m4/1.4.19#b38ced39a01e31fef5435bc634461fd2 - Cache
    meson/1.2.2#21b73818ba96d9eea465b310b5bbc993 - Cache
    ninja/1.12.1#fd583651bf0c6a901943495d49878803 - Cache
    pkgconf/2.1.0#27f44583701117b571307cf5b5fe5605 - Cache
    pkgconf/2.2.0#6462942a22803086372db44689ba825f - Cache
Resolved version ranges
    libxml2/[>=2.12.5 <3]: libxml2/2.13.4
    ninja/[>=1.10.2 <2]: ninja/1.12.1
    pkgconf/[>=2.2 <3]: pkgconf/2.2.0
    zlib/[>=1.2.11 <2]: zlib/1.3.1

======== Computing necessary packages ========
Requirements
    libiconv/1.17#1ae2f60ab5d08de1643a22a81b360c59:1cc1a9f40b0147ff70cc5ed81b8b6f860dcd0e79#48a82c042155870f1854786415673c00 - Cache
    libxml2/2.13.4#72b3bfba0c37dfe22f20c587cc468f4c:0f7e00f34df9ef6cdebb7a3d597512d17bdcb245#9419d8274aa0a0f1527b6b723592b720 - Cache
    openssl/1.1.1w#1457e590534e0b6963d32e0d5ab44eb5:42c9164af6e6754ce2521bf1e616bdbd4dd7199a#ee53900775243b93950284d72059ea30 - Cache
    xmlsec/1.3.6#57dd2c8589a0137ff8b9fd65fcb76348:9f8376c9654f60ba5cdfd7933215a71706b2152d - Build
    zlib/1.3.1#b8bc2603263cf7eccbd6e17e66b0ed76:1cc1a9f40b0147ff70cc5ed81b8b6f860dcd0e79#eae794377f2fe6b77fb16b30ee046477 - Cache
Build requirements
    autoconf/2.71#f9307992909d7fb3df459340f1932809:da39a3ee5e6b4b0d3255bfef95601890afd80709#977dcf67e8c2ef3ec98ceb1994b6a035 - Cache
    automake/1.16.5#058bda3e21c36c9aa8425daf3c1faf50:9a4eb3c8701508aa9458b1a73d0633783ecc2270#9719e51a6a62041af6a63e00eef35434 - Cache
    libtool/2.4.7#08316dad5c72c541ed21e039e4cf217b:c81087b06d1a5beef0ad711fc32d45f7a425a8f3#5431c1b6cbcb1e60dfb78b5eb59fc89a - Cache
    m4/1.4.19#b38ced39a01e31fef5435bc634461fd2:3593751651824fb813502c69c971267624ced41a#d0fee4a52cd0237a28b66e2bab404c54 - Cache
    pkgconf/2.2.0#6462942a22803086372db44689ba825f:c0b621fd4b3199fe05075171573398833dba85f4#36570ed33da9750549108235de92206e - Cache
Skipped binaries
    gnu-config/cci.20210814, meson/1.2.2, ninja/1.12.1, pkgconf/2.1.0

======== Installing packages ========
libiconv/1.17: Already installed! (1 of 10)
m4/1.4.19: Already installed! (2 of 10)
openssl/1.1.1w: Already installed! (3 of 10)
zlib/1.3.1: Already installed! (4 of 10)
pkgconf/2.2.0: Already installed! (5 of 10)
pkgconf/2.2.0: WARN: The use of 'unix_path_legacy_compat' is deprecated in Conan 2.0 and does not perform path conversions. This is retained for compatibility with Conan 1.x and will be removed in a future version.
autoconf/2.71: Already installed! (6 of 10)
libxml2/2.13.4: Already installed! (7 of 10)
automake/1.16.5: Already installed! (8 of 10)
libtool/2.4.7: Already installed! (9 of 10)
libtool/2.4.7: WARN: The use of 'unix_path_legacy_compat' is deprecated in Conan 2.0 and does not perform path conversions. This is retained for compatibility with Conan 1.x and will be removed in a future version.
libtool/2.4.7: WARN: The use of 'unix_path_legacy_compat' is deprecated in Conan 2.0 and does not perform path conversions. This is retained for compatibility with Conan 1.x and will be removed in a future version.
xmlsec/1.3.6: Calling source() in /home/amachado/.conan2/p/xmlse4a8050b5498a1/s/src
xmlsec/1.3.6: Unzipping xmlsec1-1.3.6.tar.gz to .

-------- Installing package xmlsec/1.3.6 (10 of 10) --------
xmlsec/1.3.6: Building from source
xmlsec/1.3.6: Package xmlsec/1.3.6:9f8376c9654f60ba5cdfd7933215a71706b2152d
xmlsec/1.3.6: Copying sources to build folder
xmlsec/1.3.6: Building your package in /home/amachado/.conan2/p/b/xmlse21073db9bf57b/b
xmlsec/1.3.6: Calling generate()
xmlsec/1.3.6: Generators folder: /home/amachado/.conan2/p/b/xmlse21073db9bf57b/b/build-release/conan
xmlsec/1.3.6: Generating aggregated env files
xmlsec/1.3.6: Generated aggregated env files: ['conanbuild.sh', 'conanrun.bat']
xmlsec/1.3.6: Calling build()
xmlsec/1.3.6: RUN: autoreconf --force --install
libtoolize: putting auxiliary files in '.'.
libtoolize: copying file './ltmain.sh'
libtoolize: putting macros in AC_CONFIG_MACRO_DIRS, 'm4'.
libtoolize: copying file 'm4/libtool.m4'
libtoolize: copying file 'm4/ltoptions.m4'
libtoolize: copying file 'm4/ltsugar.m4'
libtoolize: copying file 'm4/ltversion.m4'
libtoolize: copying file 'm4/lt~obsolete.m4'
configure.ac:37: installing './compile'
configure.ac:25: installing './missing'
apps/Makefile.am: installing './depcomp'

xmlsec/1.3.6: RUN: "/home/amachado/.conan2/p/b/xmlse21073db9bf57b/b/src/configure" --disable-shared --enable-static --prefix=/ '--bindir=${prefix}/bin' '--sbindir=${prefix}/bin' '--libdir=${prefix}/lib' '--includedir=${prefix}/include' '--oldincludedir=${prefix}/include' --host=i686-w64-mingw32 --build=x86_64-linux-gnu --enable-crypto-dl=no --enable-apps-crypto-dl=no --with-libxslt=no --with-openssl=yes --with-nss=no --with-nspr=no --with-gcrypt=no --with-gnutls=no --enable-mscrypto=no --enable-mscng=no --enable-docs=no --enable-mans=no --enable-pedantic=no 
checking build system type... x86_64-pc-linux-gnu
checking host system type... i686-w64-mingw32
checking for a BSD-compatible install... /usr/bin/install -c
checking whether build environment is sane... yes
checking for i686-w64-mingw32-strip... i686-w64-mingw32-strip
checking for a race-free mkdir -p... /usr/bin/mkdir -p
checking for gawk... gawk
checking whether make sets $(MAKE)... yes
checking whether make supports nested variables... yes
checking whether UID '922401112' is supported by ustar format... no
checking whether GID '922400513' is supported by ustar format... no
checking how to create a ustar tar archive... none
checking whether to enable maintainer-specific portions of Makefiles... no
checking whether make supports nested variables... (cached) yes
checking for i686-w64-mingw32-gcc... i686-w64-mingw32-gcc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.exe
checking for suffix of executables... .exe
checking whether we are cross compiling... yes
checking for suffix of object files... o
checking whether the compiler supports GNU C... yes
checking whether i686-w64-mingw32-gcc accepts -g... yes
checking for i686-w64-mingw32-gcc option to enable C11 features... none needed
checking whether i686-w64-mingw32-gcc understands -c and -o together... yes
checking whether make supports the include directive... yes (GNU style)
checking dependency style of i686-w64-mingw32-gcc... gcc3
checking how to run the C preprocessor... i686-w64-mingw32-gcc -E
checking for grep that handles long lines and -e... /usr/bin/grep
checking for egrep... /usr/bin/grep -E
checking how to print strings... printf
checking for a sed that does not truncate output... /usr/bin/sed
checking for fgrep... /usr/bin/grep -F
checking for ld used by i686-w64-mingw32-gcc... i686-w64-mingw32-ld
checking if the linker (i686-w64-mingw32-ld) is GNU ld... yes
checking for BSD- or MS-compatible name lister (nm)... i686-w64-mingw32-nm
checking the name lister (i686-w64-mingw32-nm) interface... BSD nm
checking whether ln -s works... yes
checking the maximum length of command line arguments... 1572864
checking how to convert x86_64-pc-linux-gnu file names to i686-w64-mingw32 format... func_convert_file_nix_to_w32
checking how to convert x86_64-pc-linux-gnu file names to toolchain format... func_convert_file_noop
checking for i686-w64-mingw32-ld option to reload object files... -r
checking for i686-w64-mingw32-file... no
checking for file... file
configure: WARNING: using cross tools not prefixed with host triplet
checking for i686-w64-mingw32-objdump... i686-w64-mingw32-objdump
checking how to recognize dependent libraries... pass_all
checking for i686-w64-mingw32-dlltool... i686-w64-mingw32-dlltool
checking how to associate runtime and link libraries... func_cygming_dll_for_implib
checking for i686-w64-mingw32-ar... i686-w64-mingw32-ar
checking for archiver @FILE support... @
checking for i686-w64-mingw32-strip... (cached) i686-w64-mingw32-strip
checking for i686-w64-mingw32-ranlib... i686-w64-mingw32-ranlib
checking command to parse i686-w64-mingw32-nm output from i686-w64-mingw32-gcc object... ok
checking for sysroot... no
checking for a working dd... /usr/bin/dd
checking how to truncate binary pipes... /usr/bin/dd bs=4096 count=1
checking for i686-w64-mingw32-mt... no
checking for mt... mt
checking if mt is a manifest tool... no
checking for stdio.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for strings.h... yes
checking for sys/stat.h... yes
checking for sys/types.h... yes
checking for unistd.h... yes
checking for dlfcn.h... no
checking for objdir... .libs
checking if i686-w64-mingw32-gcc supports -fno-rtti -fno-exceptions... no
checking for i686-w64-mingw32-gcc option to produce PIC... -DDLL_EXPORT -DPIC
checking if i686-w64-mingw32-gcc PIC flag -DDLL_EXPORT -DPIC works... yes
checking if i686-w64-mingw32-gcc static flag -static works... yes
checking if i686-w64-mingw32-gcc supports -c -o file.o... yes
checking if i686-w64-mingw32-gcc supports -c -o file.o... (cached) yes
checking whether the i686-w64-mingw32-gcc linker (i686-w64-mingw32-ld) supports shared libraries... yes
checking dynamic linker characteristics... Win32 ld.exe
checking how to hardcode library paths into programs... immediate
checking whether stripping libraries is possible... yes
checking if libtool supports shared libraries... yes
checking whether to build shared libraries... no
checking whether to build static libraries... yes
checking for egrep... (cached) /usr/bin/grep -E
checking for cp... /usr/bin/cp
checking for mv... /usr/bin/mv
checking for tar... /usr/bin/tar
checking for perl... /usr/bin/perl
checking for help2man... no
checking for xsltproc... /usr/bin/xsltproc
checking for man2html... no
checking for gtkdoc-mkhtml... no
checking for gtkdoc-mkdb... no
checking for gtkdoc-scan... no
checking for dirent.h that defines DIR... yes
checking for library containing opendir... none required
checking for 'i686-w64-mingw32-gcc' compiler type... "GCC"
checking for build OS... "Windows (MinGW)"
checking for development environment... disabled
checking for man pages build... disabled
checking for docs build... disabled
checking for shared library suffix... .dll
checking for __FUNCTION__ or __func__... __func__
checking pkg-config is at least version 0.9.0... yes
checking for libxml-2.0 >= 2.8.0... yes
checking for libxml2 library >= 2.8.0... yes
checking for libxslt library >= 1.0.20... disabled
checking for openssl >= 1.1.1... yes
checking for openssl library >= 1.1.1... yes (openssl version '1.1.1')
checking for nspr library >= 4.18.0... disabled
checking for nss library >= 3.35... disabled
checking for libgnutls library >= 3.6.13... disabled
checking for mscng... disabled
checking for mscrypto... disabled
checking for libgcrypt library >= 1.4.0... disabled
checking for default crypto library... yes ("openssl")
checking for Unicode support... no
checking for files support... yes
checking for FTP support... disabled
checking for HTTP support... no
checking for MD5 support... disabled
checking for RIPEMD-160 support... yes
checking for SHA1 support... yes (use discouraged)
checking for SHA224 support... yes
checking for SHA256 support... yes
checking for SHA384 support... yes
checking for SHA512 support... yes
checking for SHA3 support... yes
checking for HMAC support... yes
checking for DH support... yes
checking for DSA support... yes
checking for RSA support... yes
checking for RSA PKCS 1.5 support... yes
checking for RSA OAEP support... yes
checking for Eliptic Curve support... yes
checking for x509 support... yes
checking for DES support... yes (deprecated)
checking for AES support... yes
checking for ConcatKDF support... yes
checking for PBKDF2 support... yes
checking for GOST 2001 support... disabled
checking for GOST 2012 support... disabled
checking for XMLDSig support... yes
checking for XMLEnc support... yes
checking for mans... disabled
checking for docs... disabled
checking for apps... yes
checking for static linking... disabled
checking for xmlsec-crypto dynamic loading support... disabled
checking for xmlsec-crypto dynamic loading support in command line tool... disabled
checking for docs folder... $(htmldir)
checking for Simple Keys Manager testing... yes
checking for templates testing... yes
checking for debugging... disabled
checking for warnings as errors... disabled
checking for profiling... disabled
checking for pedantic... disabled
checking for secure memset... yes
checking that generated files are newer than configure... done
configure: creating ./config.status
config.status: creating include/xmlsec/openssl/Makefile
config.status: creating src/openssl/Makefile
config.status: creating include/xmlsec/version.h
config.status: creating Makefile
config.status: creating include/Makefile
config.status: creating include/xmlsec/Makefile
config.status: creating src/Makefile
config.status: creating apps/Makefile
config.status: creating docs/Makefile
config.status: creating docs/api/Makefile
config.status: creating man/Makefile
config.status: creating xmlsec1-openssl.pc
config.status: creating xmlsec1-gnutls.pc
config.status: creating xmlsec1-nss.pc
config.status: creating xmlsec1-gcrypt.pc
config.status: creating xmlsec1.spec
config.status: creating xmlsec1Conf.sh
config.status: creating xmlsec1-config
config.status: creating config.h
config.status: executing depfiles commands
config.status: executing libtool commands

xmlsec/1.3.6: RUN: make -j14
make  all-recursive
make[1]: Entering directory '/home/amachado/.conan2/p/b/xmlse21073db9bf57b/b/build-release'
Making all in include
make[2]: Entering directory '/home/amachado/.conan2/p/b/xmlse21073db9bf57b/b/build-release/include'
Making all in xmlsec
make[3]: Entering directory '/home/amachado/.conan2/p/b/xmlse21073db9bf57b/b/build-release/include/xmlsec'
Making all in openssl
make[4]: Entering directory '/home/amachado/.conan2/p/b/xmlse21073db9bf57b/b/build-release/include/xmlsec/openssl'
make[4]: Nothing to be done for 'all'.
make[4]: Leaving directory '/home/amachado/.conan2/p/b/xmlse21073db9bf57b/b/build-release/include/xmlsec/openssl'
make[4]: Entering directory '/home/amachado/.conan2/p/b/xmlse21073db9bf57b/b/build-release/include/xmlsec'
make[4]: Nothing to be done for 'all-am'.
make[4]: Leaving directory '/home/amachado/.conan2/p/b/xmlse21073db9bf57b/b/build-release/include/xmlsec'
make[3]: Leaving directory '/home/amachado/.conan2/p/b/xmlse21073db9bf57b/b/build-release/include/xmlsec'
make[3]: Entering directory '/home/amachado/.conan2/p/b/xmlse21073db9bf57b/b/build-release/include'
make[3]: Nothing to be done for 'all-am'.
make[3]: Leaving directory '/home/amachado/.conan2/p/b/xmlse21073db9bf57b/b/build-release/include'
make[2]: Leaving directory '/home/amachado/.conan2/p/b/xmlse21073db9bf57b/b/build-release/include'
Making all in src
make[2]: Entering directory '/home/amachado/.conan2/p/b/xmlse21073db9bf57b/b/build-release/src'
Making all in .
make[3]: Entering directory '/home/amachado/.conan2/p/b/xmlse21073db9bf57b/b/build-release/src'
  CC       app.lo
  CC       base64.lo
  CC       bn.lo
  CC       buffer.lo
  CC       c14n.lo
  CC       dl.lo
  CC       enveloped.lo
  CC       errors.lo
  CC       io.lo
  CC       keyinfo.lo
  CC       keys.lo
  CC       keysdata.lo
  CC       kw_aes_des.lo
  CC       keysmngr.lo
  CC       list.lo
  CC       membuf.lo
  CC       nodeset.lo
  CC       parser.lo
  CC       relationship.lo
  CC       strings.lo
  CC       templates.lo
  CC       transforms.lo
  CC       xmldsig.lo
  CC       xmlenc.lo
  CC       xmlsec.lo
  CC       xmltree.lo
  CC       xpath.lo
  CC       xslt.lo
  CCLD     libxmlsec1.la
make[3]: Leaving directory '/home/amachado/.conan2/p/b/xmlse21073db9bf57b/b/build-release/src'
Making all in openssl
make[3]: Entering directory '/home/amachado/.conan2/p/b/xmlse21073db9bf57b/b/build-release/src/openssl'
  CC       libxmlsec1_openssl_la-app.lo
  CC       libxmlsec1_openssl_la-ciphers.lo
  CC       libxmlsec1_openssl_la-digests.lo
  CC       libxmlsec1_openssl_la-crypto.lo
  CC       libxmlsec1_openssl_la-evp.lo
  CC       libxmlsec1_openssl_la-key_agrmnt.lo
  CC       libxmlsec1_openssl_la-kdf.lo
  CC       libxmlsec1_openssl_la-keysstore.lo
  CC       libxmlsec1_openssl_la-hmac.lo
  CC       libxmlsec1_openssl_la-kw_aes.lo
  CC       libxmlsec1_openssl_la-kw_des.lo
  CC       libxmlsec1_openssl_la-kt_rsa.lo
  CC       libxmlsec1_openssl_la-signatures.lo
  CC       libxmlsec1_openssl_la-signatures_legacy.lo
  CC       libxmlsec1_openssl_la-symkeys.lo
  CC       libxmlsec1_openssl_la-x509.lo
  CC       libxmlsec1_openssl_la-x509vfy.lo
In file included from /usr/share/mingw-w64/include/windows.h:95,
                 from /home/amachado/.conan2/p/b/xmlse21073db9bf57b/b/src/include/xmlsec/private.h:36,
                 from /home/amachado/.conan2/p/b/xmlse21073db9bf57b/b/src/src/openssl/x509.c:39:
/home/amachado/.conan2/p/b/opens49f1201400053/p/include/openssl/x509v3.h:140:9: error: expected specifier-qualifier-list before '(' token
  140 |         X509_NAME *directoryName;
      |         ^~~~~~~~~
/home/amachado/.conan2/p/b/opens49f1201400053/p/include/openssl/x509v3.h:179:5: error: expected specifier-qualifier-list before '(' token
  179 |     X509_NAME *dpname;
      |     ^~~~~~~~~
/home/amachado/.conan2/p/b/opens49f1201400053/p/include/openssl/x509v3.h:528:49: error: expected declaration specifiers or '...' before '(' token
  528 | int DIST_POINT_set_dpname(DIST_POINT_NAME *dpn, X509_NAME *iname);
      |                                                 ^~~~~~~~~
/home/amachado/.conan2/p/b/opens49f1201400053/p/include/openssl/x509v3.h:718:30: error: expected declaration specifiers or '...' before '(' token
  718 | int X509V3_NAME_from_section(X509_NAME *nm, STACK_OF(CONF_VALUE) *dn_sk,
      |                              ^~~~~~~~~
/home/amachado/.conan2/p/b/xmlse21073db9bf57b/b/src/src/openssl/private.h:40:5: error: expected specifier-qualifier-list before '(' token
   40 |     X509_NAME * subjectName;
      |     ^~~~~~~~~
/home/amachado/.conan2/p/b/xmlse21073db9bf57b/b/src/src/openssl/x509.c:1185:28: error: expected declaration specifiers or '...' before '(' token
 1185 | xmlSecOpenSSLX509NameWrite(X509_NAME* nm) {
      |                            ^~~~~~~~~
/home/amachado/.conan2/p/b/xmlse21073db9bf57b/b/src/src/openssl/x509.c: In function 'xmlSecOpenSSLKeyDataX509Write':
/home/amachado/.conan2/p/b/xmlse21073db9bf57b/b/src/src/openssl/x509.c:1330:34: warning: implicit declaration of function 'xmlSecOpenSSLX509NameWrite'; did you mean 'xmlSecOpenSSLX509SKIWrite'? [-Wimplicit-function-declaration]
 1330 |             x509Value->subject = xmlSecOpenSSLX509NameWrite(X509_get_subject_name(cert));
      |                                  ^~~~~~~~~~~~~~~~~~~~~~~~~~
      |                                  xmlSecOpenSSLX509SKIWrite
/home/amachado/.conan2/p/b/xmlse21073db9bf57b/b/src/src/openssl/x509.c:1330:32: warning: assignment to 'xmlChar *' {aka 'unsigned char *'} from 'int' makes pointer from integer without a cast [-Wint-conversion]
 1330 |             x509Value->subject = xmlSecOpenSSLX509NameWrite(X509_get_subject_name(cert));
      |                                ^
/home/amachado/.conan2/p/b/xmlse21073db9bf57b/b/src/src/openssl/x509.c:1342:35: warning: assignment to 'xmlChar *' {aka 'unsigned char *'} from 'int' makes pointer from integer without a cast [-Wint-conversion]
 1342 |             x509Value->issuerName = xmlSecOpenSSLX509NameWrite(X509_get_issuer_name(cert));
      |                                   ^
make[3]: *** [Makefile:777: libxmlsec1_openssl_la-x509.lo] Error 1
make[3]: *** Waiting for unfinished jobs....
make[3]: Leaving directory '/home/amachado/.conan2/p/b/xmlse21073db9bf57b/b/build-release/src/openssl'
make[2]: *** [Makefile:742: all-recursive] Error 1
make[2]: Leaving directory '/home/amachado/.conan2/p/b/xmlse21073db9bf57b/b/build-release/src'
make[1]: *** [Makefile:765: all-recursive] Error 1
make[1]: Leaving directory '/home/amachado/.conan2/p/b/xmlse21073db9bf57b/b/build-release'
make: *** [Makefile:583: all] Error 2

xmlsec/1.3.6: ERROR: 
Package '9f8376c9654f60ba5cdfd7933215a71706b2152d' build failed
xmlsec/1.3.6: WARN: Build folder /home/amachado/.conan2/p/b/xmlse21073db9bf57b/b/build-release
ERROR: xmlsec/1.3.6: Error in build() method, line 196
	autotools.make()
	ConanException: Error 2 while executing

```

</details>

--- a/recipes/xmlsec/all/conanfile.py
+++ b/recipes/xmlsec/all/conanfile.py
@@ -105,6 +105,9 @@ class XmlSecConan(ConanFile):
             tc = AutotoolsToolchain(self)
             if not self.options.shared:
                 tc.extra_defines.append("XMLSEC_STATIC")
+            if self.settings.os == "Windows":
+                tc.extra_defines.append("WIN32_LEAN_AND_MEAN")
+
             yes_no = lambda v: "yes" if v else "no"
             tc.configure_args.extend([
                 "--enable-crypto-dl=no",

